src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,598 @@ const ActorInit En_Box_InitVars = {
+    this->dyna.actor.minVelocityY = -50.0f;
+
+    if (globalCtx2) {} // helps the compiler store globalCtx2 into s1
+
+    if (Flags_GetTreasure(globalCtx2, this->dyna.actor.params & 0x1F)) {
+        this->alpha = 255;
+        this->iceSmokeTimer = 100;
+        EnBox_SetupAction(this, EnBox_Open);
+        this->movementFlags |= ENBOX_MOVE_STICK_TO_GROUND;
+        animFrameStart = animFrameCount;
+    } else if ((this->type == ENBOX_TYPE_SWITCH_FLAG_FALL_BIG || this->type == ENBOX_TYPE_SWITCH_FLAG_FALL_SMALL) &&
+               !Flags_GetSwitch(globalCtx2, this->switchFlag)) {
+        func_8003EBF8(globalCtx2, &globalCtx2->colCtx.dyna, this->dyna.dynaPolyId);
+        if (Math_Rand_ZeroOne() < 0.5f) {
+            this->movementFlags |= ENBOX_MOVE_FALL_ANGLE_SIDE;
+        }
+        this->unk_1A8 = -12;
+        EnBox_SetupAction(this, EnBox_FallOnSwitchFlag);
+        this->alpha = 0;
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->dyna.actor.flags |= 0x10;
+    } else if ((this->type == ENBOX_TYPE_ROOM_CLEAR_BIG || this->type == ENBOX_TYPE_ROOM_CLEAR_SMALL) &&
+               !Flags_GetClear(globalCtx2, this->dyna.actor.room)) {
+        EnBox_SetupAction(this, EnBox_AppearOnRoomClear);
+        func_8003EBF8(globalCtx2, &globalCtx2->colCtx.dyna, this->dyna.dynaPolyId);
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y - 50.0f;
+        this->alpha = 0;
+        this->dyna.actor.flags |= 0x10;
+    } else if (this->type == ENBOX_TYPE_9 || this->type == ENBOX_TYPE_10) {
+        EnBox_SetupAction(this, func_809C9700);
+        this->dyna.actor.flags |= 0x2000000;
+        func_8003EBF8(globalCtx2, &globalCtx2->colCtx.dyna, this->dyna.dynaPolyId);
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y - 50.0f;
+        this->alpha = 0;
+        this->dyna.actor.flags |= 0x10;
+    } else if (this->type == ENBOX_TYPE_SWITCH_FLAG_BIG && !Flags_GetSwitch(globalCtx2, this->switchFlag)) {
+        EnBox_SetupAction(this, EnBox_AppearOnSwitchFlag);
+        func_8003EBF8(globalCtx2, &globalCtx2->colCtx.dyna, this->dyna.dynaPolyId);
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y - 50.0f;
+        this->alpha = 0;
+        this->dyna.actor.flags |= 0x10;
+    } else {
+        if (this->type == ENBOX_TYPE_4 || this->type == ENBOX_TYPE_6) {
+            this->dyna.actor.flags |= 0x80;
+        }
+        EnBox_SetupAction(this, EnBox_WaitOpen);
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->movementFlags |= ENBOX_MOVE_STICK_TO_GROUND;
+    }
+
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.posRot.rot.y += 0x8000;
+    this->dyna.actor.posRot.rot.z = this->dyna.actor.shape.rot.z;
+    this->dyna.actor.initPosRot.rot.z = this->dyna.actor.shape.rot.z;
```

Comment:
```suggestion
    this->dyna.actor.posRot.rot.y += 0x8000;
    this->dyna.actor.initPosRot.rot.z = this->dyna.actor.posRot.rot.z = this->dyna.actor.shape.rot.z = 0;
```

Reply:
Somehow that made the stack usage go up, but removing the `linkAge` temp thanks to `(0, gSaveContext.linkAge)` brought it back to matching. Nice